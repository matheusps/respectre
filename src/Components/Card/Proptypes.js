import { string } from 'prop-types'

export default {
  modifiers: string,
}
