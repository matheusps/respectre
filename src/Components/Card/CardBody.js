import React from 'react'
import Proptypes from '../Proptypes'

const CardBody = ({ children, modifiers, ...props }) => (
    <div className={`card-body ${modifiers}`} {...props} >
        { children }
    </div>
)

CardBody.propTypes = Proptypes

export default CardBody