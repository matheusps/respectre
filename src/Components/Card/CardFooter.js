import React from 'react'
import Proptypes from './Proptypes'

const CardFooter = ({ children, modifiers, ...props }) => (
  <div className={`card-footer ${modifiers}`} {...props}>
    {children}
  </div>
)

CardFooter.propTypes = Proptypes

export default CardFooter
