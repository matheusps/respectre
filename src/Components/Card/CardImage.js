import React from 'react'
import Proptypes from '../Proptypes'

const CardImage = ({ src, modifiers, children, ...props }) => (
    <div className={`card-image ${modifiers}`} {...props}>
        <img src={src} className="img-responsive" />
    </div>
)

CardImage.propTypes = Proptypes

export default CardImage