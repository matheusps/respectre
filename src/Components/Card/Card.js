import React from 'react'
import PropTypes from './Proptypes'
import CardHeader from './Header';
import CardBody from './CardBody';
import CardImage from './CardImage';
import CardFooter from './CardFooter';

const Card = ({ children, modifiers, ...props }) => (
    <div className={`card ${modifiers}`} {...props}>
        { children }
    </div>
)

Card.Header = CardHeader
Card.Body = CardBody
Card.Image = CardImage
Card.Footer = CardFooter

Card.propTypes = PropTypes

export default Card