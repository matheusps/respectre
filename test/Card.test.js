import React from 'react'
import Card from '../src/Components/Card'
import renderer from 'react-test-renderer'

test('Card Matches Snapshot', () => {
  const component = renderer.create(
    <Card>
      <Card.Image src="https://static1.squarespace.com/static/58817d60d2b85782338743f1/t/59acb965f9a61e6ef12a8021/1504491926588/gis+header+1.jpg?format=1500w" />
      <Card.Header>
        <Card.Header.Title modifiers="h5">My title</Card.Header.Title>
        <Card.Header.Subtitle modifiers="text-gray">
          My subtitle
        </Card.Header.Subtitle>
      </Card.Header>
      <Card.Body>This describes my card body</Card.Body>
      <Card.Footer>
        <span>Footer</span>
      </Card.Footer>
    </Card>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
