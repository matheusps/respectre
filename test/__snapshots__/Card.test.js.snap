// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match snapshot 1`] = `
<div
  className="card undefined"
>
  <div
    className="card-image undefined"
  >
    <img
      className="img-responsive"
      src="testing"
    />
  </div>
  <div
    className="card-header undefined"
  >
    <div
      className="card-title h5"
    >
      My title
    </div>
    <div
      className="card-title text-gray"
    >
      My subtitle
    </div>
  </div>
  <div
    className="card-body undefined"
  >
    This describes my card body
  </div>
  <div
    className="card-footer undefined"
  >
    <span>
      Footer
    </span>
  </div>
</div>
`;
